hazelcast/src/test/java/com/hazelcast/cluster/TcpIpHostnameJoinTest.java

```diff
@@ -80,9 +81,14 @@ public void test_whenMembersDefinedViaIpAndHostnameMix() {
     }
 
     @Test
-    public void test_whenClusterNamesDifferSecondMemberFailsToStart() {
-        instance("cluster1", HOSTNAME1);
-        assertThrows(IllegalStateException.class, () -> instance("cluster2", HOSTNAME2));
+    public void test_whenClusterNamesDiffer_then_have_two_separate_clusters() {
+        HazelcastInstance cluster1 = instance("cluster1", HOSTNAME1);
+        HazelcastInstance cluster2 = instance("cluster2", HOSTNAME2);
+
+        assertTrueAllTheTime(() -> {
+            assertClusterSize(1, cluster1);
+            assertClusterSize(1, cluster2);
+        }, 20);
```

Comment:
Maybe the time allocation is too generous. Wouldn't be 5-10sec enough?

Reply:
15 seconds is `connectionTimeoutSeconds` in this test file. This was the reason. Also this is a slow test.